File: PPWC/Minor_Assignment_6/Q4.c
#include<stdio.h>
int f(int n, int k){
    if(n==0) return 0;
    else if(n%2) return f(n/2, 2*k)+k;
    else return f(n/2, 2*k)-k;
}
int main(){
    printf("%d",f(20,1));
    return 0;
}

/*
Output--
9
*/

File: Python/Major_Assignment_2/Q4.py
def supplies_data(supplies):
    sup={}
    for item in supplies:
        sup[item[0]]={item[1]:item[2]}
    return sup

supplies=eval(input("Enter Supplies: "))
print("Output:\n",supplies_data(supplies))

'''
Output--
Enter Supplies: [("Kyiv", "Food", 500), ("Moscow", "Medicines", 200), ("Lviv", "Water", 300), ("Saint Petersburg","Blankets", 100), ("Kharkiv", "Food", 150)]
Output:
 {'Kyiv': {'Food': 500}, 'Moscow': {'Medicines': 200}, 'Lviv': {'Water': 300}, 'Saint Petersburg': {'Blankets': 100}, 'Kharkiv': {'Food': 150}}
 '''